addons/mail/static/src/core/common/thread_model.js

```diff
@@ -834,26 +836,35 @@ export class Thread extends Record {
      * @param {import("models").Message} [messageId] if not provided, load around newest message
      */
     async loadAround(messageId) {
-        if (!this.messages.some(({ id }) => id === messageId)) {
+        if (
+            this.status === "loading" ||
+            (this.isLoaded && this.messages.some(({ id }) => id === messageId))
```

Comment:
If you are close to end of a load-around attempt, it sucks that it doesn't behave consistently by loading around the same amount of messages in all cases.

If I load 30 messages and I load around to the former last message, I expect to load around 30 messages there, not doing nothing like this new implementation does. What I don't like with this inconsistent behaviour is that it forces to play around, e.g. the `fetchNewMessages()` that means something different from `loadAround()`, which sucks because this makes the code more complicated than it should.

Reply:
I don't get your point with "playing around". `fetchNewMessages` doesn't do anything passed the first load for channels. That's not the case for `loadAround` since it can be done several time in a thread life so they are really different in that matter. On the other hand I agree that this condition on `status` suck. It's only there because `onMounted` is triggered twice in some specific cases, leading to duplicate `loardAround` rpcs.